File: pilha/Pilha Estática/Java/Pilha Dupla/EmpilhavelDupla.java
public interface EmpilhavelDupla {
	//métodos da Pilha1
	void empilhar1(Object dado);
	void atualizar1(Object dado);
	Object desempilhar1();
	Object espiar1();	
	boolean estaCheia1(); //igual a estaCheia2()
	boolean estaVazia1();
	String imprimir1();
	
	//métodos da Pilha2
	void empilhar2(Object dado);
	void atualizar2(Object dado);
	Object desempilhar2();
	Object espiar2();	
	boolean estaCheia2(); //igual a estaCheia1()
	boolean estaVazia2();
	String imprimir2();
}

File: pilha/Pilha Estática/Kotlin/Pilha Dupla/EmpilhavelDupla.kt
public interface EmpilhavelDupla {
	//métodos da Pilha1
    fun empilhar1(dado: Any?)
    fun desempilhar1(): Any?
    fun topo1(): Any?
    fun estaCheia1(): Boolean
    fun estaVazia1(): Boolean
    fun imprimir1(): String	
	
	//métodos da Pilha2
    fun empilhar2(dado: Any?)
    fun desempilhar2(): Any?
    fun topo2(): Any?
    fun estaCheia2(): Boolean  //igual a estaCheia1()
    fun estaVazia2(): Boolean
    fun imprimir2(): String	
}